### DISCUSSION TITLE
salesforce DAG example with REST and BULK API

### DISCUSSION THREAD

**Discussion by jayantsande25** [apache/airflow#23668] [Q&A]
Hi @mariotaddeucci & Friends, 

I saw your example DAG. But looks like its in connection with Tableau. 
Can anyone write a sample DAG for Salesforce(Rest API) to 
Query rows from object and WRITE(Update) rows of Object(Table) in Salesforce using Python?

I am referring your example here
https://github.com/apache/airflow/blob/0a3ff43d41d33d05fb3996e61785919effa9a2fa/airflow/providers/salesforce/example_dags/example_salesforce_apex_rest.py

**Comment by mariotaddeucci** [comment]
Hey @jayantsande25
This example is to call salesforce apex endpoint.
The salesforce hook uses the [simple-salesforce](https://github.com/simple-salesforce/simple-salesforce) to handle the connection and call the methods. On amazon provider there's an salesforce to s3 operator, that brings the data to amazon S3.

Based on simple-salesforce docs, you can run bulk api by doing

        sf_hook = SalesforceHook()
        sf = sf_hook.get_conn()
        sf.bulk.TheObjectName.update([ {"Id":"foo", "another_field": "bar"} ])

You can make it dynamically using `getattr` function

Eg:

        action = "delete"
        sobject "account"
        data = [{"Id": "foo"}]

        sf_hook = SalesforceHook()
        sf = sf_hook.get_conn()

        bulk = getattr(getattr(sf.bulk, sobject), action)

        bulk(data)
